runtime/jcl/common/jdk_internal_misc_ScopedMemoryAccess.cpp

```diff
@@ -86,6 +86,9 @@ Java_jdk_internal_misc_ScopedMemoryAccess_closeScope0(JNIEnv *env, jobject insta
 					setHaltFlag(walkThread, J9_PUBLIC_FLAGS_CLOSE_SCOPE);
 					walkThread->scopedError = errorObj;
 					walkThread->closeScopeObj = closeScopeObj;
+
+					Assert_JCL_true(vm->closeScopeNotifyCount >= 0);
+					vm->closeScopeNotifyCount += 1;
```

Comment:
Im not sure we need a global count either. Is it possible for us to have a notify without a preceeding wait? Even if it is, notifying an empty list isnt incorrect.

Reply:
- the signal can be lost if a notify is sent without a preceding wait leading to a deadlock.
- the global counter reduces the number of notifies and amount of work to decide whether to continue in closeScope0. the extra notifies and work can cause hurdles in scaling when CPU resources are scarce. i believe the counter is more efficient.

Reply:
> the signal can be lost if a notify is sent without a preceding wait leading to a deadlock.

Im not sure I understand how this scenario can occur. Notifies are only thrown in the handler, and the handler can only run if the event was sent, nnd the code always waits if an event is sent, which means that there must have been a preceeding wait.

Reply:
>  the extra notifies and work can cause hurdles in scaling when CPU resources are scarce.

The global atomic counter is a problem in and of itself as the number of CPUs grows

Reply:
Im not opposed to optimizing this, but I believe its unnecesarry as this code should only run if badly written code is being run (ie closing a scope that someone else is using). Any code that relies on exceptions for code flow cannot be efficient. I expect closecope0 to not block 99.99% of the time, if this isnt true its the applications fault.

I'd still like to understand why the counter is needed for correctness.